deps/rabbitmq_mqtt/src/rabbit_mqtt_reader.erl

```diff
@@ -24,19 +24,21 @@
 
 -include("rabbit_mqtt.hrl").
 
+-type option(T) :: undefined | T.
+
 -define(HIBERNATE_AFTER, 1000).
 -define(PROTO_FAMILY, 'MQTT').
 
 -record(state,
-        {socket,
-         proxy_socket :: undefined | {rabbit_proxy_soket, any(), any()},
+        {socket :: rabbit_net:socket(),
+         proxy_socket :: option({rabbit_proxy_socket, any(), any()}),
          await_recv :: boolean(),
-         deferred_recv :: undefined | binary(),
-         parse_state,
+         deferred_recv :: option(binary()),
+         parse_state :: atom(),
```

Comment:
The `parse_state` can also be a tuple in https://github.com/rabbitmq/rabbitmq-server/blob/1fd4a6d353bd57793d38f7640bdfce1c6e6ada64/deps/rabbitmq_mqtt/src/rabbit_mqtt_frame.erl#L32 . Ideally, we should also declare the parser state to be opaque.

Reply:
done